### PR TITLE
Feature(UX): Connect to API

### PR DESCRIPTION
**Pull request by blessedjasonmwanza**
## Feature(UX): Connect to API
In this user experience update:
- Refactor the redux code
- Use Redux Thunk middleware to make async requests to the API.
- Update UI based on API data

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,22 +1,49 @@
...
+    const [bookEntries] = value;
+    return {
+      id: key,
+      ...bookEntries,
+    };
+  });
+  dispatch({
+    type: FETCH_BOOKS,
+    payload: bookList,
+  });
```
Comment (hiromataba) [inline src/redux/books/books.js:36]: - [x] Good job on  adding, deleting, and fetching your books to the API but when rendering your books there is a bit of issue, When I go to the categories page it works fine I see the message `under construction but when I navigate back to books they duplicate themselves but if I reload the page the duplication disappears, Kindly have a look at that video and fix the issue :point_down: 

https://user-images.githubusercontent.com/75126481/151522207-21c1c5cc-13cd-4e61-843f-7b6547dde398.MP4

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi :wave:  @blessedjasonmwanza , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### To highlight :1st_place_medal: 
- Nice PR title and description :heavy_check_mark: 
- Nice README file :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [x] [OPTIONAL] You have a good README file however I think it would be much better to remove some of the `available script` that was generated by `CRA` and please add a command to install npm `npm install

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (blessedjasonmwanza) [reply src/redux/books/books.js:36]: Hello @hiromataba ,

Thanks a million for spotting the issue 😄 . 

I have successfully managed to fix the issue and will go on request a final review.

PS - I have also added the install script following your guideline.

> HappyCoding 🤓

Comment (hiromataba) [reply src/redux/books/books.js:36]: Thank you @blessedjasonmwanza :pray:

**Review by uwadonat — APPROVED** [review]
Hi **@blessedjasonmwanza**, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**